### ISSUE TITLE
ec_deployment.tags showing up in plan when unspecified

### ISSUE THREAD

**Issue by matschaffer** [elastic/terraform-provider-ec#334]
## Description

In trying out the minimal example, I found that after initial apply I was left with a `tags = {}` diff on next plan.

```
  # ec_deployment.example_minimal has been changed
  ~ resource "ec_deployment" "example_minimal" {
        id                     = "REDACTED"
        name                   = "my_example_deployment"
      + tags                   = {}
        # (7 unchanged attributes hidden)




        # (4 unchanged blocks hidden)
    }
```

I was tempted to add it to the docs, but probably better if the provider would consider `tags = {}` to be equivalent to unspecified tags in the code.

HCL to reproduce:

```hcl
provider "ec" {
}

terraform {
  required_version = "1.2.7"
  required_providers {
    ec = {
      source = "elastic/ec"
      version = "0.4.1"
    }
  }
}


data "ec_stack" "latest" {
  version_regex = "latest"
  region        = "us-east-1"
}

resource "ec_deployment" "example_minimal" {
  # Optional name.
  name = "my_example_deployment"

  # Mandatory fields
  region                 = "us-east-1"
  version                = data.ec_stack.latest.version
  deployment_template_id = "aws-io-optimized-v2"

  # Use the deployment template defaults
  elasticsearch {}

  kibana {}
}
```

**Comment by marclop** [comment]
@matschaffer mind specifying the terraform version and provider you used? this will be helpful later on to make sure we're reproducing it with the proper Terraform versions.

**Comment by matschaffer** [comment]
Oh, sorry about that! I added a full HCL that should work to reproduce.

**Comment by marclop** [comment]
I haven't had the chance to look into this in detail but this didn't happen pre `1.0.0`, so I'm assuming something changed with the introduction of that Terraform version.

**Comment by dimuon** [comment]
Need to check - it can be already solved.

**Comment by matschaffer** [comment]
I updated the description and retried. No outstanding changes anymore so guess it got fixed somewhere between 0.2.1 and 0.4.1.

Thanks all!
